recurring_content_detector/detector.py

```diff
@@ -1,51 +1,40 @@
-import os
-import itertools
-import operator
-import numpy as np
-import pickle
-import faiss
-import datetime
+from os.path import join, isfile, dirname
+from os import makedirs, listdir
+from itertools import groupby
+from operator import itemgetter
+from numpy import array, float32, percentile, vstack, append
+from pickle import load
+from faiss import IndexFlatL2
+from datetime import timedelta
```

Comment:
And is there a reason you changed the import style? Because I think it makes the code harder to read unfortunately, in this way it's harder to see of which module a specific function originates.